airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/DogStatsDMetricSingleton.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
+import com.timgroup.statsd.StatsDClient;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Light wrapper around the DogsStatsD client to make using the client slightly more ergonomic.
+ * <p>
+ * This class mainly exists to help Airbyte instrument/debug application on Airbyte Cloud.
+ * <p>
+ * Open source users are free to turn this on and consume the same metrics.
+ */
+@Slf4j
+public class DogStatsDMetricSingleton {
+
+  private static DogStatsDMetricSingleton instance;
+  private final StatsDClient statsDClient;
+  private final boolean instancePublish;
+
+  public static synchronized DogStatsDMetricSingleton getInstance() {
+    if (instance == null) {
+      throw new RuntimeException("You must initialize configuration with the initialize() method before getting an instance.");
+    }
+    return instance;
+  }
+
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish) {
+    initialize(app, publish, null, null);
+  }
+
+  @VisibleForTesting
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish, final String ddAgentHost, final String ddPort) {
+    if (instance != null) {
+      throw new RuntimeException("You cannot initialize configuration more than once.");
+    }
+
+    if (!publish) {
+      log.info("Starting stub DogstatsD client..");
+      instance = new DogStatsDMetricSingleton();
+      return;
+    }
+
+    log.info("Starting DogStatsD client..");
+    if (!ddAgentHost.isBlank() || !ddPort.isBlank()) {
+      instance = new DogStatsDMetricSingleton(app.getApplicationName(), publish, ddAgentHost, ddPort);
+      return;
+    }
+
+    instance = new DogStatsDMetricSingleton(app.getApplicationName(), publish);
+  }
+
+  @VisibleForTesting
+  public synchronized static void flush() {
+    instance = null;
+  }
+
+  private DogStatsDMetricSingleton(final String appName, final boolean publish) {
+    instancePublish = publish;
+    statsDClient = new NonBlockingStatsDClientBuilder()
+        .prefix(appName)
+        .hostname(System.getenv("DD_AGENT_HOST")) // Matches Airbyte Cloud Datadog env vars.
```

Comment:
why are we parsing env variables here? is there a reason we wouldn't pass this through `EnvConfigs` like we do with other env variables? In october there was a big refactor to make sure we weren't paring env variables anywhere in the code base but centralize it all through `EnvConfigs` and push it to the top of the application. I want to make sure we don't revert those efforts.

Reply:
good question. my initial thoughts were to place this in the constructor for convenience and also expose a flexible constructor to allow injecting (keeping our clean up in mind). I was also thinking of 'hiding' the variables for monitoring since they are cloud specific.

on second thought, no reason to keep this away from OSS users. we can put it in the env vars like you pointed out and document this for OSS folks with Datadog. How is that?

Reply:
made the changes - looks much cleaner now

Reply:
nice!